cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/InMemoryFlowProgramRunner.java

```diff
@@ -167,10 +170,12 @@ private ProgramOptions createFlowletOptions(String name, int instanceId, int ins
     systemArgs.put(ProgramOptionConstants.INSTANCE_ID, Integer.toString(instanceId));
     systemArgs.put(ProgramOptionConstants.INSTANCES, Integer.toString(instances));
 
-    return new SimpleProgramOptions(name, new BasicArguments(systemArgs), options.getUserArguments());
+    ProgramId originalProgramId = options.getProgramId();
```

Comment:
Why not? FlowSpec has the `Map<String, FlowletDefinitions>`, where the String is the flowlet name. The flowlet name is different from the flow name right?